dotnet/src/SemanticKernel.Core/Data/VolatileVectorRecordStore.cs

```diff
@@ -0,0 +1,170 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Data;
+
+/// <summary>
+/// Service for storing and retrieving vector records, that uses an in memory dictionary as the underlying storage.
+/// </summary>
+/// <typeparam name="TRecord">The data model to use for adding, updating and retrieving data from storage.</typeparam>
+[Experimental("SKEXP0001")]
+public class VolatileVectorRecordStore<TRecord> : IVectorRecordStore<string, TRecord>
```

Comment:
Do we want to allow inheriting from the class? Same for the VolatileVectorRecordStoreOptions class.

Reply:
Thanks, good catch.  Sealed them.